Test/dafny0/ResolutionErrors.dfy

```diff
@@ -2875,3 +2875,19 @@ module MapDisjointnessNoMore {
     assert a.Keys !! b.Keys;  // instead, this is the way to do it
   }
 }
+
+// --------------- expect statements ------------------------------
+
+module ExpectStatements {
+
+  function UnsafeDivide(a: int, b: int): int {
+    expect b != 0;  // expect statement is not allowed in this context
+    a / b
+  }
+
+  method M() {
+    ghost var g := 5;
+    expect forall i : int :: i == i;  // error: quantifiers in non-ghost contexts must be compilable
+    expect false, if g == 5 then "boom" else "splat"; // 
+  }
```

Comment:
Add two more tests: one where the message is ghost (which isn't allowed) and one where the message has a non-string type.

Reply:
The first case is already covered by line 2891 - the message depends on `g` which is ghost. I neglected to fill in the comment with "error: ghost variables are allowed only in specification contexts" so I'll do that.

I'll add the second case, but to be clear the message argument can be any value, which is string-ified in the same way as any argument passed to `print`. That is, I'll definitely add the test but it will be one that succeeds - probably a clone of `ExpectWithMessage.dfy`